odoo/fields.py

```diff
@@ -691,11 +692,11 @@ def get_description(self, env):
 
     @property
     def _description_searchable(self):
-        return bool(self.store or self.search)
+        return bool(self.store or self.search or self.adhoc)
 
     @property
     def _description_sortable(self):
-        return (self.column_type and self.store) or (self.inherited and self.related_field._description_sortable)
+        return (self.column_type and self.store) or (self.inherited and self.related_field._description_sortable) or self.adhoc
```

Comment:
adhoc is the only change is fields definition. Potentially could be avoided by using ``_generate_adhoc_fields``